bin/docserv-stitch

```diff
@@ -110,30 +112,57 @@ for i in "$@"
       ;;
       --simplify)
         simplify_config=1
+        shift
```

Comment:
At the top, we still use `for i in $@`, but if you're using `shift` here, then the list of parameters will be modified while looping over it. Maybe that works in practice, I have not tried.
Conceptually, it does not make sense though: Either use `while` + `shift` or `for` + and no shifting.

I am not against using `while`. However, I am not sure **why** you are changing the parameter handling so much. I always had the impression that it worked.

Reply:
I've tried to add the new option, but that didn't work. Not sure why. So I've tried this approach with shift to make the change as minimal as possible.

Maybe we should even consider `getopt`. I've created a short test script which I think addresses your concerns about usability:

```shell
ME="${0##*/}"

export POSIXLY_CORRECT=1
ARGS=$(getopt -o "h" -l "help,simplify,revalidate-only,spinner,valid-languages:,site-sections:" -n "$ME" -- "$@")

eval set -- "$ARGS"

while true; do
    case "$1" in
        -h|--help|--simplify|--revalidate-only|--spinner)
            echo "Found option $1"
            shift
            ;;
        --valid-languages|--site-sections)
            echo "Found option $1 with value '$2'"
            shift 2
            break
            ;;
        --)
            shift
            break
            ;;
        --*)
            echo "Found unknown option $1"
            shift
            ;;
    esac
done

unset POSIXLY_CORRECT
```

If I run it, I get:

```
$ ./test-parsing.sh -h --spinner -x -y --valid-languages="de"
test-parsing.sh: invalid option -- x
test-parsing.sh: invalid option -- y
Found option -h
Found option --spinner
Found option --valid-languages with value 'de'
```

This will find all unknown options at once, but can also deal with the `=` atrocities.